mirascope/core/mistral/stream.py

```diff
@@ -26,10 +29,10 @@ class MistralStream(
     BaseStream[
         MistralCallResponse,
         MistralCallResponseChunk,
-        ChatMessage,
-        ChatMessage,
-        ChatMessage,
-        ChatMessage,
+        AssistantMessage | SystemMessage | ToolMessage | UserMessage,
```

Comment:
so these were originally all the same because there was only one message type, but now we should use the correct message types for the generics (see [here](https://github.com/Mirascope/mirascope/blob/9202d7f32fffa3798c95c3051d9e98a953eab27c/mirascope/core/base/stream.py#L55)):

1. _UserMessageParamT = UserMessage
2. _AssistantMessageParamT = AssistantMessage
3. _ToolMessageParamT = ToolMessage
4. _MessageParamT = SystemMessage | UserMessage | AssistantMessage | ToolMessage